bakery/service.go

```diff
@@ -296,7 +296,9 @@ func Discharge(key *KeyPair, checker ThirdPartyChecker, id string) (*macaroon.Ma
 		return nil, nil, errgo.Mask(err, errgo.Any)
 	}
 	// Note that the discharge macaroon does not need to
-	// be stored persistently.
+	// be stored persistently. Indeed, it would be a problem if
```

Comment:
As discussed online, this test is here in part to protect against that bug possibility.